src/view/renderer.js

```diff
@@ -418,24 +428,77 @@ export default class Renderer {
 	 * @private
 	 */
 	_updateSelection() {
-		// If there is no selection - remove it from DOM elements that belongs to the editor.
+		// If there is no selection - remove DOM and fake selections.
 		if ( this.selection.rangeCount === 0 ) {
 			this._removeDomSelection();
+			this._removeFakeSelection();
 
 			return;
 		}
 
-		if ( !this.isFocused ) {
+		const domRoot = this.domConverter.getCorrespondingDomElement( this.selection.editableElement );
+
+		// Do nothing if there is no focus, or there is no DOM element corresponding to selection's editable element.
+		if ( !this.isFocused || !domRoot ) {
 			return;
 		}
 
-		const selectedEditable = this.selection.editableElement;
-		const domRoot = this.domConverter.getCorrespondingDomElement( selectedEditable );
+		// Render selection.
+		if ( this.selection.isFake ) {
+			this._updateFakeSelection( domRoot );
+		} else {
+			this._removeFakeSelection();
+			this._updateDomSelection( domRoot );
+		}
+	}
+
+	/**
+	 * Updates fake selection.
+	 *
+	 * @private
+	 * @param {HTMLElement} domRoot Valid DOM root where fake selection container should be added.
+	 */
+	_updateFakeSelection( domRoot ) {
+		const domDocument = domRoot.ownerDocument;
+
+		// Create fake selection container if one does not exist.
+		if ( this._fakeSelectionContainer === null ) {
+			this._fakeSelectionContainer = domDocument.createElement( 'div' );
+			this._fakeSelectionContainer.style.position = 'fixed';
+			this._fakeSelectionContainer.style.top = 0;
+			this._fakeSelectionContainer.style.left = '-1000px';
+		}
+
+		// Add fake container if not already added.
+		if ( this._fakeSelectionContainer.parentElement === null ) {
+			domRoot.appendChild( this._fakeSelectionContainer );
+		}
 
-		if ( !domRoot ) {
-			return;
+		// Update contents.
+		const content = this.selection.fakeSelectionLabel || '&nbsp;';
+
+		if ( content !== this._fakeSelectionContainer.innerHTML ) {
+			this._fakeSelectionContainer.innerHTML = content;
```

Comment:
Imagine what happens if someone set alt of their image to `"<img onerror=alert(1)>"`. That alt will be used to represent the image o it will be passed straight to `innerHTML`. Better to create a text node and set its data.